registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/validator/LostRidRequestValidator.java

```diff
@@ -0,0 +1,205 @@
+				"LostRidRequestValidator::validate()::exit");
+
+	}
+
+	/**
+	 * Validate id.
+	 *
+	 * @param id the id
+	 * @throws RegStatusAppException the reg status app exception
+	 */
+	private void validateId(String id) throws RegStatusAppException {
+		LostRidValidationException exception = new LostRidValidationException();
+
+		if (Objects.isNull(id)) {
+
+			throw new RegStatusAppException(PlatformErrorMessages.RPR_RGS_MISSING_INPUT_PARAMETER_ID, exception);
+		} else if (!env.getProperty(REG_LOSTRID_SERVICE_ID).equals(id)) {
+
+			throw new RegStatusAppException(PlatformErrorMessages.RPR_RGS_INVALID_INPUT_PARAMETER_ID, exception);
+
+		}
+	}
+
+
+	/**
+	 * Validate filter.
+	 *
+	 * @param version the version
+	 * @param errors  the errors
+	 * @return true, if successful
+	 * @throws RegStatusAppException
+	 * 
+	 */
+	private void validateFilter(List<FilterInfo> filterInfos) throws WorkFlowSearchException, RegStatusAppException {
+		LostRidValidationException exception = new LostRidValidationException();
+		if (Objects.isNull(filterInfos)) {
+			throw new RegStatusAppException(PlatformErrorMessages.RPR_RGS_MISSING_INPUT_PARAMETER_VERSION, exception);
+		}
+		for (FilterInfo filter : filterInfos) {
+			if (filter.getColumnName().equals("name") || filter.getColumnName().equals("email")
+					|| filter.getColumnName().equals("phone") || filter.getColumnName().equals("centerId")
+					|| filter.getColumnName().equals("registrationDate")) {
+
+			} else {
+				throw new RegStatusAppException(PlatformErrorMessages.RPR_RGS_MISSING_INPUT_PARAMETER_VERSION, exception);
+			}
+		}
+
+	}
+
+	/**
+	 * Validate ver.
+	 *
+	 * @param ver the ver
+	 * @throws RegStatusAppException the reg status app exception
+	 */
+	private void validateVersion(String ver) throws RegStatusAppException {
+		String version = env.getProperty(REG_LOSTRID_APPLICATION_VERSION);
+		LostRidValidationException exception = new LostRidValidationException();
+		if (Objects.isNull(ver)) {
+			throw new RegStatusAppException(PlatformErrorMessages.RPR_RGS_MISSING_INPUT_PARAMETER_VERSION, exception);
+
+		} else if (!version.equals(ver)) {
+
+			throw new RegStatusAppException(PlatformErrorMessages.RPR_RGS_INVALID_INPUT_PARAMETER_VERSION, exception);
+		}
+	}
+
+	/**
+	 * Validate sortInfo.
+	 *
+	 * @param version the version
+	 * @param errors  the errors
+	 * @return true, if successful
+	 * @throws RegStatusAppException
+	 */
+	private void validateSortField(List<SortInfo> sortInfos) throws WorkFlowSearchException, RegStatusAppException {
+		LostRidValidationException exception = new LostRidValidationException();
+		if (Objects.isNull(sortInfos)) {
+			throw new RegStatusAppException(PlatformErrorMessages.RPR_RGS_MISSING_INPUT_PARAMETER, exception);
+		} else if (sortInfos.get(0).getSortField().equals("createDateTime")
+				|| sortInfos.get(0).getSortField().equals("updateDateTime")
```

Comment:
Why should we allow sort on fields that we don't allow filters

Reply:
becoz if someone want to sort on  sync createddatetime then it should give the result

Reply:
Then you should give that value also rt for them to see ?

Reply:
taken care removed createDateTime field